## Manatee Mortality Stats from Florida Fish and Wildlife - PDF format

In [1]:
import pandas as pd

### Convert PDF to spreadsheet using [this website](https://altoconvertpdftoexcel.com/)

In [5]:
df = pd.read_excel('manatees_2018.xlsx', header=None)
df.head()

0   1   2   3   4    5   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS... NaN NaN NaN NaN  NaN   
1                                             County NaN NaN NaN NaN  NaN   
2                                             Citrus NaN NaN NaN NaN  NaN   
3                                               Levy NaN NaN NaN NaN  NaN   
4                                            Broward NaN NaN NaN NaN  NaN   

                    6    7         8    9   ...   16   17              18  \
0                  NaN  NaN       NaN  NaN  ...  NaN  NaN             NaN   
1                 Date  NaN  Field ID  NaN  ...  NaN  NaN            City   
2  2018-02-01 00:00:00  NaN  MNW18021  NaN  ...  NaN  NaN   Crystal River   
3  2018-02-01 00:00:00  NaN  MNW18026  NaN  ...  NaN  NaN       Cedar Key   
4  2018-02-01 00:00:00  NaN  MSE18024  NaN  ...  NaN  NaN  Ft. Lauderdale   

    19   20                                   21   22   23  24  25  
0  NaN  NaN                                  NaN  NaN  NaN NaN NaN  
1  NaN  NaN                       Probable Cause  NaN  NaN NaN NaN  
2  NaN  NaN                       Natural: Other  NaN  NaN NaN NaN  
3  NaN  NaN                 Natural: Cold Stress  NaN  NaN NaN NaN  
4  NaN  NaN  Human Related: Watercraft Collision  NaN  NaN NaN NaN  

[5 rows x 26 columns]

In [8]:
df.tail(20)

0   1   2   3   4   \
735                                             Walton NaN NaN NaN NaN   
736                                          St. Lucie NaN NaN NaN NaN   
737                                                Lee NaN NaN NaN NaN   
738                                             Monroe NaN NaN NaN NaN   
739                                            Collier NaN NaN NaN NaN   
740                                            Collier NaN NaN NaN NaN   
741                                                NaN NaN NaN NaN NaN   
742                                                NaN NaN NaN NaN NaN   
743                                                NaN NaN NaN NaN NaN   
744                             For The Period of Time NaN NaN NaN NaN   
745                                 Total=         824 NaN NaN NaN NaN   
746  Mortality comparisons for the time period from... NaN NaN NaN NaN   
747                                               Year NaN NaN NaN NaN   
748                                               2018 NaN NaN NaN NaN   
749                                               2017 NaN NaN NaN NaN   
750                                               2016 NaN NaN NaN NaN   
751                                               2015 NaN NaN NaN NaN   
752                                               2014 NaN NaN NaN NaN   
753                                               2013 NaN NaN NaN NaN   
754                                     5-year average NaN NaN NaN NaN   

             5                    6                7         8   \
735         NaN  2018-12-30 00:00:00              NaN  MNW18166   
736         NaN  2018-12-30 00:00:00              NaN  MSE18098   
737         NaN  2018-12-30 00:00:00              NaN  MSW18340   
738         NaN  2018-12-30 00:00:00              NaN  MSW18341   
739         NaN  2018-12-30 00:00:00              NaN  MSW18343   
740         NaN  2018-12-31 00:00:00              NaN  MSW18342   
741         NaN                  NaN              NaN       NaN   
742         NaN                  NaN              NaN       NaN   
743         NaN                  NaN              NaN       NaN   
744         NaN                  NaN              NaN       NaN   
745         NaN                  NaN              NaN       NaN   
746         NaN                  NaN              NaN       NaN   
747  Watercraft                  NaN  Flood Gate/Lock       NaN   
748   123 (15%)                  NaN                6       NaN   
749   111 (21%)                  NaN                3       NaN   
750   106 (20%)                  NaN                2       NaN   
751    86 (21%)                  NaN                5       NaN   
752    69 (19%)                  NaN                3       NaN   
753     73 (9%)                  NaN                5       NaN   
754    89 (18%)                  NaN                3       NaN   

                           9   ...           16       17          18  \
735                       NaN  ...          NaN      NaN   Niceville   
736                       NaN  ...          NaN      NaN  Ft. Pierce   
737                       NaN  ...          NaN      NaN  Cape Coral   
738                       NaN  ...          NaN      NaN    Flamingo   
739                       NaN  ...          NaN      NaN     Ochopee   
740                       NaN  ...          NaN      NaN      Naples   
741                       NaN  ...          NaN      NaN         NaN   
742                       NaN  ...          NaN      NaN         NaN   
743                       NaN  ...          NaN      NaN         NaN   
744  01/01/2018 -\n12/31/2018  ...          NaN      NaN         NaN   
745                       NaN  ...          NaN      NaN         NaN   
746                       NaN  ...          NaN      NaN         NaN   
747                       NaN  ...  Cold Stress  Natural         NaN   
748                       NaN  ...           79      228         NaN   
749      

In [10]:
# test slice

df.iloc[749:753, 0:]

0   1   2   3   4          5    6  7    8    9   ...  16  17   18  19  \
749  2017 NaN NaN NaN NaN  111 (21%)  NaN  3  NaN  NaN  ...  30  93  NaN  26   
750  2016 NaN NaN NaN NaN  106 (20%)  NaN  2  NaN  NaN  ...  24  86  NaN  36   
751  2015 NaN NaN NaN NaN   86 (21%)  NaN  5  NaN  NaN  ...  18  44  NaN  31   
752  2014 NaN NaN NaN NaN   69 (19%)  NaN  3  NaN  NaN  ...  26  29  NaN  17   

      20   21  22   23  24  25  
749  114  NaN  29  538 NaN NaN  
750  113  NaN  34  520 NaN NaN  
751   95  NaN  24  405 NaN NaN  
752   80  NaN  38  371 NaN NaN  

[4 rows x 26 columns]

In [14]:
clean_df = df.iloc[:741, 0:]
clean_df.tail(10)

0   1   2   3   4    5                    6    7         8   \
731  Hillsborough NaN NaN NaN NaN  NaN  2018-12-27 00:00:00  NaN  MNW18164   
732  Indian River NaN NaN NaN NaN  NaN  2018-12-28 00:00:00  NaN  MEC18150   
733  Hillsborough NaN NaN NaN NaN  NaN  2018-12-29 00:00:00  NaN  MNW18165   
734       Broward NaN NaN NaN NaN  NaN  2018-12-29 00:00:00  NaN  MSE18097   
735        Walton NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MNW18166   
736     St. Lucie NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSE18098   
737           Lee NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSW18340   
738        Monroe NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSW18341   
739       Collier NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSW18343   
740       Collier NaN NaN NaN NaN  NaN  2018-12-31 00:00:00  NaN  MSW18342   

      9   ...   16   17            18   19   20  \
731  NaN  ...  NaN  NaN  Apollo Beach  NaN  NaN   
732  NaN  ...  NaN  NaN     Sebastian  NaN  NaN   
733  NaN  ...  NaN  NaN         Tampa  NaN  NaN   
734  NaN  ...  NaN  NaN     Hollywood  NaN  NaN   
735  NaN  ...  NaN  NaN     Niceville  NaN  NaN   
736  NaN  ...  NaN  NaN    Ft. Pierce  NaN  NaN   
737  NaN  ...  NaN  NaN    Cape Coral  NaN  NaN   
738  NaN  ...  NaN  NaN      Flamingo  NaN  NaN   
739  NaN  ...  NaN  NaN       Ochopee  NaN  NaN   
740  NaN  ...  NaN  NaN        Naples  NaN  NaN   

                                      21   22   23  24  25  
731                 Human Related: Other  NaN  NaN NaN NaN  
732         Undetermined: Too Decomposed  NaN  NaN NaN NaN  
733                       Natural: Other  NaN  NaN NaN NaN  
734         Undetermined: Too Decomposed  NaN  NaN NaN NaN  
735                 Natural: Cold Stress  NaN  NaN NaN NaN  
736         Undetermined: Too Decomposed  NaN  NaN NaN NaN  
737                Perinatal (<= 150 cm)  NaN  NaN NaN NaN  
738  Human Related: Watercraft Collision  NaN  NaN NaN NaN  
739                 Natural: Cold Stress  NaN  NaN NaN NaN  
740         Undetermined: Too Decomposed  NaN  NaN NaN NaN  

[10 rows x 26 columns]

#### Drop NA columns

In [15]:
clean_df.isnull().sum()

0       0
1     741
2     741
3     741
4     741
5     741
6       1
7     741
8       1
9     741
10    741
11    741
12      1
13      3
14    741
15      1
16    741
17    741
18      1
19    741
20    741
21      1
22    741
23    741
24    741
25    741
dtype: int64

In [22]:
# drop columns with a threshold of more than 3 NA values because I'm not sure what the column 13 is about

clean_df.dropna(axis=1, thresh=3, inplace=True)
clean_df.head()

0                    6   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS...                  NaN   
1                                             County                 Date   
2                                             Citrus  2018-02-01 00:00:00   
3                                               Levy  2018-02-01 00:00:00   
4                                            Broward  2018-02-01 00:00:00   

         8    12         13                 15              18  \
0       NaN  NaN        NaN                NaN             NaN   
1  Field ID  Sex  Size (cm)           Waterway            City   
2  MNW18021    M        269          Kings Bay   Crystal River   
3  MNW18026    M        202  Black Point Swamp       Cedar Key   
4  MSE18024    U        275    Port Everglades  Ft. Lauderdale   

                                    21  
0                                  NaN  
1                       Probable Cause  
2                       Natural: Other  
3                 Natural: Cold Stress  
4  Human Related: Watercraft Collision

In [ ]:
# drop first row and convert second row to headers

clean_df.drop()

#### Can drop rows 744 & 745 because they're just a summary of the deaths for all of 2018:
* 824 total deaths
* 123 Watercraft Deaths (Florida only)

### Mortality comparisons for the time period from 01/01 to the ending date (final + preliminary data)


In [13]:
comparison_df = df.iloc[747:, 0:]
comparison_df

0   1   2   3   4           5    6                7    8   \
747            Year NaN NaN NaN NaN  Watercraft  NaN  Flood Gate/Lock  NaN   
748            2018 NaN NaN NaN NaN   123 (15%)  NaN                6  NaN   
749            2017 NaN NaN NaN NaN   111 (21%)  NaN                3  NaN   
750            2016 NaN NaN NaN NaN   106 (20%)  NaN                2  NaN   
751            2015 NaN NaN NaN NaN    86 (21%)  NaN                5  NaN   
752            2014 NaN NaN NaN NaN    69 (19%)  NaN                3  NaN   
753            2013 NaN NaN NaN NaN     73 (9%)  NaN                5  NaN   
754  5-year average NaN NaN NaN NaN    89 (18%)  NaN                3  NaN   

      9   ...           16       17   18           19            20   21  \
747  NaN  ...  Cold Stress  Natural  NaN  Unrecovered  Undetermined  NaN   
748  NaN  ...           79      228  NaN          105           120  NaN   
749  NaN  ...           30       93  NaN           26           114  NaN   
750  NaN  ...           24       86  NaN           36           113  NaN   
751  NaN  ...           18       44  NaN           31            95  NaN   
752  NaN  ...           26       29  NaN           17            80  NaN   
753  NaN  ...           40      198  NaN          100           125  NaN   
754  NaN  ...           27       90  NaN           42           105  NaN   

                     22     23  24  25  
747  Undetermined Other  Total NaN NaN  
748                  29    824 NaN NaN  
749                  29    538 NaN NaN  
750                  34    520 NaN NaN  
751                  24    405 NaN NaN  
752                  38    371 NaN NaN  
753                 148    830 NaN NaN  
754                  54    532 NaN NaN  

[8 rows x 26 columns]